posthog/hogql/database.py

```diff
@@ -48,17 +48,17 @@ def get_field(self, name: str) -> DatabaseField:
     def clickhouse_table(self):
         raise NotImplementedError("Table.clickhouse_table not overridden")
 
-    def get_asterisk(self) -> List[str]:
-        list: List[str] = []
-        for field in self.__fields__.values():
+    def get_asterisk(self) -> Dict[str, DatabaseField]:
```

Comment:
this will need to support person on events conditionally when it's enabled 😬 (i.e. if there's a person.properties, make a decision whether to join persons table or go to the person_properties column instead).

Unclear how this should work yet, but not blocking now!

Reply:
Also food for thought: Many of our query optimisations are based on programmatically choosing which columns to select (example: choosing event.properties can OOM if it's not actually necessary for the subsequent query because these are big json payloads).

Would be nice if hogql could work with that logic, even for random client queries.

Reply:
Yeah, person properties will get their own handling. Part of it is happening in https://github.com/PostHog/posthog/pull/14286 . Right now we've come sort of full circle. In that PR you can _only_ get person properties from a joined table, not from the person_properties field :D. Since PoE is off for most everyone, I'm trying to get this working first. It should be easy to say "don't join a table, take a field instead"